Jenkinsfile

```diff
@@ -129,31 +131,68 @@ def getTests(String baseUrl) {
 }
 
 def getScenarios(String baseUrl, String testId){
-    def scenarios = [[Id: "-1", Name: "Default"]]
-    def isScenarioChoosen = "${params['Run with scenarios']}".toBoolean()
-    if (isScenarioChoosen) {
+    def scenarios = [[Id: "0", Name: "Default"]]
+    def isScenarioChoosen = "${params['Run with datasets']}".toBoolean()
+    def isScenarioPresentForTest = scenariosPresent(baseUrl, testId)
+    if (isScenarioChoosen && isScenarioPresentForTest) {
         def jsonSlurper = new JsonSlurper()
         URL url = new URL("${baseUrl}/projects/${params['Project']}/tests/${testId}/scenarios")
-        HttpURLConnection conn = url.openConnection()
-        conn.setRequestMethod("GET")
-        conn.setDoInput(true)
-        conn.setRequestProperty("Authorization", "APIKEY $useMangoApiKey")
-        conn.connect()
-        String content
-        def responseCode = conn.responseCode
-        if (responseCode == 200) {
-            InputStream inputStream = conn.getInputStream()
-            content = inputStream.getText()
-        } else {
-            throw new Exception("Scenarios get request failed with code: ${responseCode}")
-        }
+        String content = getWebResponseContent(url, "Scenarios")
         def scenarioPage = jsonSlurper.parseText(content)
         if (scenarioPage != null) {
             scenarioPage.each { scenario ->
-                echo "${scenario.Name}"        
+                echo "${scenario.Name}"
                 scenarios << [Id: scenario.Id, Name: scenario.Name]
             }
         }
+        return scenarios;
+    }
+    return null
+}
+
+def getAllScenarioIds(ArrayList<LinkedHashMap<String, String>> scenarioList) {
+    def listOfIds = []
+    scenarioList.each { item ->
+        listOfIds << item.Id
+    }
+    return listOfIds;
+}
+
+def addQueryParameterToUrl(String path, Map<String, Object> queryParams) {
+    if (queryParams.isEmpty()) {
+        return new URL(path)
+    }
+    path = path + "?"
+    for (final def keyValue in queryParams.entrySet()) {
+        def query = keyValue.getKey()
+        def queryValue = keyValue.getValue()
+        if (queryValue instanceof ArrayList) {
+            queryValue.each { value ->
+                path += "${query}=${URLEncoder.encode(value as String, 'UTF-8')}&"
+            }
+        } else {
+            path += "${query}=${URLEncoder.encode(queryValue as String, "UTF-8")}&"
+        }
+    }
+    return new URL(path.substring(0, path.length() - 1))
+}
+
+boolean scenariosPresent(String baseUrl, String testId) {
+    def jsonSlurper = new JsonSlurper()
+    URL url = new URL("${baseUrl}/projects/${params['Project']}/tests/${testId}")
+    String content = getWebResponseContent(url, "Dataset");
+    def scenarioPage = jsonSlurper.parseText(content)
+    return scenarioPage["Parameters"].size() >= 1
+}
+
+def getWebResponseContent(URL url, String requestedFor) {
```

Comment:
Can we return JSON response since all the methods using this anyway parse the response?

Reply:
Also we need a better name for this function, need to signify that it's actually making a request of some sort.